docker/DOCKER_FULL_EXAMPLE.md

```diff
@@ -0,0 +1,117 @@
+## Docker Full example setup
+
+This page shows an example setup consisting of several docker containers working together for easy deployment.
+
+The main documentation on how to use the bluesky container is in our main [README](../README.md)
+
+### Assumptions for this example
+
+We will be using these docker containers:
+- [mysql:5.7](https://hub.docker.com/_/mysql/)
+- [sphen/bluesky](https://hub.docker.com/r/sphen/bluesky/)
+- [abiosoft/caddy](https://hub.docker.com/r/abiosoft/caddy/)
+
+These are the values being used below - and will need modification from you:
+
+| Setting | Value |
+| --- | --- |
+| MySQL root password | admin |
+| Server FQDN | bluesky.example.com |
+| Web admin pass | admin |
+| Email alert address | email@example.com |
+| SMTP Server | smtp.office365.com:587 |
+| SMTP user | email@example.com |
+| SMTP pass | yourpassword |
+
+These are the local directories for persistent data:
+```
+/var/docker/bluesky/db
+/var/docker/bluesky/certs
+/var/docker/bluesky/admin.ssh
+/var/docker/bluesky/bluesky.ssh
+/var/docker/caddy
+```
+
+### Steps
+
+#### Create the local storage directories
+
+> _Note you may want to set more secure permissions_
+
+```
+mkdir -p /var/docker/bluesky/db \
+  /var/docker/bluesky/certs \
+  /var/docker/bluesky/admin.ssh \
+  /var/docker/bluesky/bluesky.ssh \
+  /var/docker/caddy
+```
+
+#### Create MySQL container
+
+> _Note that the MySQL root password is in this command. If you use a complex password with offending characters you should enclose the password in ''. Passwords do not work with a \ in them_
+
+```
+docker run -d --name bluesky_db \
+  -v /var/docker/bluesky/db:/var/lib/mysql \
+  -e MYSQL_ROOT_HOST=% \
+  -e MYSQL_ROOT_PASSWORD=admin \
+  --restart always \
+  mysql/mysql-server:5.7
+```
+
+**Wait a minute or two for the MySQL container to initialize.** You can get the status of the container by running `docker ps -a`.  Wait until you see it with a status of **(healthy)**
+
+#### Create BlueSky container
+
+> **Note:** _All the variables that need to change.  If you use complex passwords with offending characters you should enclose the password in single quotes.  Passwords do not work with a backslash in them_
+
+> **Note:** _We do not care about SSL certs here as the Caddy container will take care of that_
+
+```
+docker run -d --name bluesky \
```

Comment:
Ah, here they are.  I missed these on the first trial run.

Reply:
yes so in this first pass (i can do more in the future) i just focussed on adapting what you did for docker.  that includes not really worrying about SSL much (no auto LetsEncrypt).  I decided in the "full setup" example to walk the user through creating a caddy container which would provide SSL automatically for the bluesky container.  in the case that the user did not want to do this and wanted to use their own certs i have a way to pass the cert and private key to the container which is have tested and works.